### PR TITLE
fix: safari hls

### PR DESCRIPTION
**Pull request by BrandonOCasey**

### CODE REVIEW

File: src/plugin.js [hunk 1]
```diff
@@ -1,31 +1,18 @@
...
-import rgbFragmentShader from './rgb-fragment-shader';
-import rgbaFragmentShader from './rgba-fragment-shader';
-import vertexShader from './vertex-shader';
+import * as utils from './utils';
 
 // import controls so they get regisetered with videojs
 import './cardboard-button';
 import './big-vr-play-button';
 
-const validProjections = [
```
Comment (BrandonOCasey) [inline src/plugin.js]: moved to utils

File: src/plugin.js [hunk 2]
```diff
@@ -46,51 +33,6 @@ const errors = {
   }
 };
 
-const getInternalProjectionName = function(projection) {
```
Comment (BrandonOCasey) [inline src/plugin.js]: moved to utils

File: src/plugin.js [hunk 3]
```diff
@@ -115,8 +57,9 @@ class VR extends Plugin {
       player.errors({errors});
     }
 
-    // we need this as IE 11 reports that it has a VR display, but isnt compatible with Video as a Texture. for example
-    if (videojs.browser.IE_VERSION) {
+    // IE 11 does not support enough webgl to be supported
+    // older safari does not support cors, so it wont work
+    if (videojs.browser.IE_VERSION || !utils.corsSupport) {
```
Comment (BrandonOCasey) [inline src/plugin.js]: older version of safari that don't support cors at all (<9 wont work)

File: src/plugin.js [hunk 4]
```diff
@@ -366,46 +323,41 @@ class VR extends Plugin {
...
-      this.videoTexture.flipY = false;
-    } else if (isHLS() && videojs.browser.IS_SAFARI) {
-      this.log('Safari + HLS = flipY hack');
-      this.videoTexture.format = THREE.RGBFormat;
-      this.videoTexture.flipY = false;
-    } else {
-      this.videoTexture.format = THREE.RGBFormat;
-    }
+    // https://bugs.webkit.org/show_bug.cgi?id=179417
+    if (videojs.browser.IS_ANY_SAFARI && utils.isHLS(this.player_.currentSource().type)) {
```
Comment (BrandonOCasey) [inline src/plugin.js]: previously we had a hack to get video textures working here, this uses the canvas image texture instead as that will make ios 10 and 11 both work

Comment (BrandonOCasey) [reply src/plugin.js]: From my testing on an IPAD It doesn't perform bad enough that I think its a bad solution.

**Comment by pagarwal123** [comment]
QA LGTM

**Pull request merged**